addons/account/models/account_tax.py

```diff
@@ -669,7 +669,7 @@ def recompute_base(base_amount, incl_tax_amounts):
             sum_repartition_factor = sum(tax_repartition_lines.mapped('factor'))
 
             #compute the tax_amount
-            if price_include and i in cached_tax_amounts:
+            if price_include and i in cached_tax_amounts and sum_repartition_factor != 0:
```

Comment:
I think the problem is more in `recompute_base`:
1)
Change `incl_tax_amounts['percent_taxes'].append((i, tax.amount * sum_repartition_factor))`
to `incl_tax_amounts['percent_taxes'].append((i, tax.amount, sum_repartition_factor))`
2) Make the computation by scaling the amounts with `sum_repartition_factor`
3) cache the unfactorized tax amount instead of the factorized version.
4) remove your diff in this line